api/exec/kubernetes_deploy.go

```diff
@@ -123,6 +124,8 @@ func (deployer *KubernetesDeployer) command(operation string, userID portainer.U
 
 	var stderr bytes.Buffer
 	cmd := exec.Command(command, args...)
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, "POD_NAMESPACE=default")
```

Comment:
Would a better way to do it be like the following?

Otherwise, seems like a good change since the POD with the agent should really always be in the default ns

```
Change:
    if namespace != "" {
		args = append(args, "--namespace", namespace)
    }
To:
     if namespace == "" {
		namespace = "default"
    }
    
    args = append(args, "--namespace", namespace)
```

Reply:
@hookenz   when namespace == "", users can specify the namespace in the manifest file too. So we can not override it by assignment namespace = "default".